docs/design/adrs/2024-05-02-enable-project-versioning

```diff
@@ -0,0 +1,63 @@
+
+## Context and Problem Statement
+
+Versioning allows users and developers to track changes between deployments, either by comparing code differences between versions or by referring to a changelog.
+
+It also signals to users the nature of the changes, such as breaking changes, new functionality, or bug fixes, through the use of [semantic versioning](https://semver.org/).
+
+Adopting versioning will give us more control over releases and improve communication to our users through changelogs and migration guides.
+
+## Decision Drivers
+
+* Minimal process (it should be straightforward to publish a release)
+* Small time overhead (steps should be automated where possible)
+* Low maintenance solution
+
+## Considered Options
+
+* GitHub Releases
+
+## Decision Outcome
+
+TBD
+
+## Pros and Cons of the Options
+
+### GitHub Releases
+
+GitHub Releases is a feature which can be enabled directly in the repository, and allows you to create releases with release notes, @mentions of contributors, and link assets such as binaries.
+
+Pros:
+
+* **Supported solution**: GitHub Releases is built-in to GitHub repositories, so is well supported and low overhead to enable. It is used by many product teams, e.g. [vscode](https://github.com/microsoft/vscode), [semantic-kernel](https://github.com/microsoft/semantic-kernel), [WSL](https://github.com/microsoft/WSL), etc.
+
+* **Easy publishing**: You can create, edit and delete releases using the browser or Releases API, which makes it simple to use.
+
+* **Built-in features**: You can compare releases from the repository, see [Comparing releases](https://docs.github.com/en/repositories/releasing-projects-on-github/comparing-releases). You can also see the latest release from the repository homepage.
+
+* **Automation**: There is support to automatically generate [release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes) based on changes since the previous release, it also includes who has contributed.
+
+* **Simple**: There are extensive [docs](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases).
+
+Cons:
+
+* **Automatic releases**: Whilst it is straight-forward to publish a release manually, there are extra steps required if we would want to have automatic releases.
+    - This [GitHub Blog post](https://github.blog/2021-12-16-5-automations-every-developer-should-be-running/#5-automate-your-releases-and-release-notes-with-github-actions) shares how you can use GitHub Actions to run a workflow which will create a release.
+    - If we were to automate the release, we would need to make decisions on the release schedule or trigger, for example, should a new release be created each time a change is merged to `main`? Or should it be done on a schedule such as daily, weekly, etc?
```

Comment:
Releases for each commit could be somewhat problematic/unnecessary for things like dependabot updates

Reply:
See my latest updates for a suggestion here, which should allow us to bundle a bunch of the dependabot updates into a single release if we wished to.